org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/ClientPreferences.java

```diff
@@ -36,26 +39,34 @@ public boolean isSignatureHelpSupported(){
 	}
 
 	public boolean isCompletionSnippetsSupported() {
-		return v3supported && capabilities.getTextDocument().getCompletion().getCompletionItem().getSnippetSupport();
+		//@formatter:off
+		return v3supported && capabilities.getTextDocument().getCompletion() != null
+				&& capabilities.getTextDocument().getCompletion().getCompletionItem() != null
+				&& isTrue(capabilities.getTextDocument().getCompletion().getCompletionItem().getSnippetSupport());
+		//@formatter:on
 	}
 
 	public boolean isV3Supported() {
 		return v3supported;
 	}
 
 	public boolean isFormattingDynamicRegistrationSupported() {
-		return v3supported && capabilities.getTextDocument().getFormatting().getDynamicRegistration();
+		return v3supported && isDynamicRegistrationSupported(capabilities.getTextDocument().getFormatting());
```

Comment:
While this will fix the case where the last fileld in the chain is null, it does not solve the general case, because getTextDocument() could already return null. I know this is kind of rubbish API, but there you go.